bofire/strategies/doe/objective.py

```diff
@@ -474,6 +488,205 @@ def _convert_input_to_tensor(
+                domain, model, n_space, delta, transform_range=None
+            )
+            constraints = constraints_as_scipy_constraints(
+                domain, n_space, ignore_nchoosek=True
+            )
+            bounds = nchoosek_constraints_as_bounds(domain, n_space)
+            if ipopt_options is None:
+                ipopt_options = {}
+            _ipopt_options = {"maxiter": 500, "disp": 0}
+            for key in ipopt_options.keys():
+                _ipopt_options[key] = ipopt_options[key]
+            if _ipopt_options["disp"] > 12:
+                _ipopt_options["disp"] = 0
+
+            result = minimize_ipopt(
+                objective.evaluate,
+                x0=x0,
+                bounds=bounds,
+                constraints=standardize_constraints(constraints, x0, "SLSQP"),
+                options=_ipopt_options,
+                jac=objective.evaluate_jacobian,
+            )
+
+            design = pd.DataFrame(
+                result["x"].reshape(n_space, len(domain.inputs)),
+                columns=domain.inputs.get_keys(),
+                index=[f"exp{i}" for i in range(n_space)],
+            )
+        else:
+            low, high = domain.inputs.get_bounds(specs={})
+            points = [
+                list(
+                    np.linspace(
+                        low[i],
+                        high[i],
+                        int(100 * (high[i] - low[i])),
+                    )
+                )
+                for i in range(len(low))
+            ]
+            points = np.array(list(product(*points)))
+            points = pd.DataFrame(points, columns=domain.inputs.get_keys())
+            if len(domain.constraints) > 0:
+                fulfilled = domain.constraints(experiments=points)
+                fulfilled = np.array(
+                    [
+                        np.array(fulfilled.iloc[:, i]) <= 0.0
+                        for i in range(fulfilled.shape[1])
+                    ]
+                )
+                fulfilled = np.array(np.prod(fulfilled, axis=0), dtype=bool)
+                design = points[fulfilled]
+            else:
+                design = points
+            n_space = len(design)
+            print(
+                f"Filling the design space with {len(design)} equally spaced grid points."
+            )
+
+        try:
+            domain.validate_candidates(
+                candidates=design.apply(lambda x: np.round(x, 8)),
+                only_inputs=True,
+                tol=1e-4,
+            )
+        except (ValueError, ConstraintNotFulfilledError):
+            warnings.warn(
+                "Some points do not lie inside the domain or violate constraints. Please check if the \
+                    results lie within your tolerance.",
+                UserWarning,
+            )
+
+        model_formula = get_formula_from_string(
+            model_type=model, rhs_only=True, domain=domain
+        )
```

Comment:
As far as I understood it correctly, until here, a lot of functionality is just recoding other stuff that to come up with a starting set. And this recoding is partially done to prevent, circular imports. What do you think about adding an additional keyword argument to __init__, which just takes the inital design as input. Then the user also has much more flexibility in controlling this critical part of the design. Furtheremore, it would make the code much cleaner. What do you think?

Reply:
In the DoEStrategy, we could then add this part of coming up with an inital setup in case somebody wants to perform an I-optimal design. But here, the user would still have the option todo custom stuff.

What do you think about a tidy-up of the DoE functionality in general (after this PR ;))?

Reply:
We  could start to revamp `find_local_max_iptopt`, by changing the signature in a way that it just accepts the instantiated `objective` class, which is holding the `domain` instead also taking the 'domain` as argument. The model type would then also be only part of the objective and not also of the `find_local_max_ipopt` signature. This way, we can make it all much cleaner and modular. What do you think?